src/chainstate/coordinator/mod.rs

```diff
@@ -845,12 +1196,156 @@ impl<
+                        //                            | PROCESS RC P  |   | PROCESS RC Q |   | PROCESS RC R |
+                        if let Some(ancestor_info) = self.process_exit_reward_cycle(
+                            stacks_block_height_in_cycle,
+                            &parent_exit_info,
+                        )? {
+                            info!("RCPR: cc: ancestor - rc is greater than parent rc");
+                            // Check if there is some proposal. If so, need to check for a veto.
+                            if let Some(curr_exit_proposal) = ancestor_info.curr_exit_proposal {
+                                let veto_passed = self.read_veto_state(
+                                    ancestor_info.block_reward_cycle,
+                                    curr_exit_proposal,
+                                    &stacks_block_id,
+                                )?;
+                                // if veto fails, record exit block height
+                                if !veto_passed {
+                                    info!("RCPR: cc: veto did not pass for {}", curr_exit_proposal);
+                                    current_exit_at_rc = Some(curr_exit_proposal);
+                                } else {
+                                    info!("RCPR: cc: storing veto info for {}", curr_exit_proposal);
+                                    // record the veto in invalid_reward_cycles
+                                    let sortdb_handle = self
+                                        .sortition_db
+                                        .tx_handle_begin(&canonical_sortition_tip)?;
+                                    invalid_reward_cycles.push(curr_exit_proposal);
+                                    sortdb_handle.commit()?;
+                                }
+                            } else {
+                                // tally votes of previous reward cycle if there is no veto happening
+                                // if there is consensus for some proposal, record it
+                                let tallied_proposal = self.tally_votes(
+                                    ancestor_info.block_reward_cycle,
+                                    ancestor_info.curr_exit_at_reward_cycle,
+                                    ancestor_info.invalid_reward_cycles,
+                                    &stacks_block_id,
+                                    &block_receipt.header.consensus_hash,
+                                )?;
+                                if let Some(exit_proposal) = tallied_proposal {
+                                    info!(
+                                        "RCPR: cc: storing proposal info for {:?}",
+                                        exit_proposal
+                                    );
+                                    // record the proposal in invalid_reward_cycles
+                                    let sortdb_handle = self
```

Comment:
Wanna combine these transactions?

Reply:
Actually, why is this first transaction necessary?  It seems that it's not actually committing anything.